src/PolyhedralGeometry/Cone/constructors.jl

```diff
@@ -72,7 +73,7 @@ end
 
 
 @doc Markdown.doc"""
-    positive_hull([::Type{T} = fmpq,] R::Union{Oscar.MatElem, AbstractMatrix, SubObjectIterator})
+    positive_hull([::Type{T} = fmpq,] R::MatrixUnion)
 
 A polyhedral cone, not necessarily pointed, defined by the positive hull of the
 rows of the matrix `R`. This means the cone consists of all positive linear
```

Comment:
So to remove this confusion, again some documentation for `MatrixUnion` would help. Starting with the mathematical description is a good idea and later stating how the input would be interpreted if the input is of specific format.

Allowing `AbstractVector{AbstractVector}`s etc is also an aim for this PR, but this is still to do. Though it might work already to some extent, I am not sure at the moment.

Also, if i read your last paragraph correctly, you suggest to have two types, e.g. `RayCollection` and `PointCollection`, instead of a single `MatrixUnion`? Under the aspect that `SomeMatrix == AnyVecOrMat` this would probably mean that `RayCollection == PointCollection`, but this is not necessarily bad, because it still shows the user the correct usage (and passing rays into a point input would still throw an error hen converting at the beginning). We then have to be consequential and have something like `HalfplaneCollection` and `HyperplaneCollection` where the current `MatrixUnion` or `TupleUnion` describes neither points nor rays.